Algorithms/Stack/EvaluatePostfixExpression.txt

```diff
@@ -0,0 +1,77 @@
+// Evaluate Postfix Expression
+// Language Used: JAVA
+
+// Given a postfix expression, the task is to evaluate the expression and print the final value. Operators will only include the basic // arithmetic operators like *,/,+ and - . 
+
+// Input format: The first line of input will contains an integer T denoting the no of test cases . Then T test cases follow. Each //	         test case contains an postfix expression.
+
+// Output format: For each test case, in a new line, evaluate the postfix expression and print the value.
+
+// Sample Input :
+// 2
+// 231*+9-
+// 123+*8-
+
+
+// Sample Output:
+// -4
+// -3
+
+
+
+import java.util.*;
+
+class EvaluatePostfixExpression{
+    
+    public static int evaluate(int first,int second,char operator)
+    {
+        switch(operator)
+        {
+            case '+':
+                return first+second;
+            case '-':
+                return second-first;
+            case '*':
+                return first*second;
+            case '/':
+                return second/first;
+            case '^':
+                int result=1;
+                for(int i=1;i<first;i++)
+                {
+                    result*=second;
+                }
+                return result;
+            default:
+                return 0;
+        }
+        
+    }
+    
+	public static void main (String[] args) 
+	{
+		Scanner scan = new Scanner(System.in);
```

Comment:
a more simpler approach could be used using stack...this takes more time complexity

Reply:
I think this is the simplest approach as it's time complexity O(n) for each test case.